From 98a52fbbbc302a2d387e535026a969c11382c9c0 Mon Sep 17 00:00:00 2001
From: Kevin Sheppard <kevin.k.sheppard@gmail.com>
Date: Sat, 14 Apr 2018 21:01:11 +0100
Subject: [PATCH] COMPAT: Add forward compat for is_list_like

Pandas 0.23 moves is_list_like
Add import to compat
Ignore UnstableAPIWarnings in Google
Swap tslib.Timestamp for Timestamp
---
 pandas_datareader/compat/__init__.py          |  7 +++++++
 pandas_datareader/fred.py                     |  3 ++-
 pandas_datareader/tests/google/test_google.py | 19 +++++++++++--------
 5 files changed, 23 insertions(+), 10 deletions(-)

diff --git a/pandas_datareader/compat/__init__.py b/pandas_datareader/compat/__init__.py
index 58496335..46c68b78 100644
--- a/pandas_datareader/compat/__init__.py
+++ b/pandas_datareader/compat/__init__.py
@@ -14,6 +14,7 @@
 PANDAS_0190 = (PANDAS_VERSION >= LooseVersion('0.19.0'))
 PANDAS_0200 = (PANDAS_VERSION >= LooseVersion('0.20.0'))
 PANDAS_0210 = (PANDAS_VERSION >= LooseVersion('0.21.0'))
+PANDAS_0230 = (PANDAS_VERSION >= LooseVersion('0.23.0'))
 
 if PANDAS_0190:
     from pandas.api.types import is_number
@@ -39,6 +40,12 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
     from pandas.util.testing import assertRaisesRegexp as assert_raises_regex
     get_filepath_or_buffer = com.get_filepath_or_buffer
 
+if PANDAS_0230:
+    from pandas.core.dtypes.common import is_list_like
+else:
+    from pandas.core.common import is_list_like
+
+
 if compat.PY3:
     from urllib.error import HTTPError
 else:
diff --git a/pandas_datareader/fred.py b/pandas_datareader/fred.py
index c1eba812..666c4223 100644
--- a/pandas_datareader/fred.py
+++ b/pandas_datareader/fred.py
@@ -1,4 +1,5 @@
-from pandas.core.common import is_list_like
+from pandas_datareader.compat import is_list_like
+
 from pandas import concat, read_csv
 
 from pandas_datareader.base import _BaseReader
diff --git a/pandas_datareader/tests/google/test_google.py b/pandas_datareader/tests/google/test_google.py
index 538f2090..f94be60c 100644
--- a/pandas_datareader/tests/google/test_google.py
+++ b/pandas_datareader/tests/google/test_google.py
@@ -16,6 +16,9 @@
 import pandas.util.testing as tm
 from pandas.util.testing import assert_series_equal
 
+pytestmark = pytest.mark.filterwarnings(
+    'ignore::pandas_datareader.exceptions.UnstableAPIWarning')
+
 
 def assert_n_failed_equals_n_null_columns(wngs, obj, cls=SymbolWarning):
     all_nan_cols_dict = {}
@@ -64,8 +67,8 @@ def test_google(self):
         end = datetime(2013, 1, 27)
         for locale in self.locales:
             with tm.set_locale(locale):
-                    panel = web.DataReader("NYSE:F", 'google', start, end)
-                    assert panel.Close[-1] == 13.68
+                panel = web.DataReader("NYSE:F", 'google', start, end)
+                assert panel.Close[-1] == 13.68
 
             with pytest.raises(Exception):
                 web.DataReader('NON EXISTENT TICKER', 'google', start, end)
@@ -153,9 +156,9 @@ def test_get_multi2(self):
     def test_dtypes(self):
         # see gh-3995, gh-8980
         data = web.get_data_google(
-                'NYSE:F',
-                start='JAN-01-10',
-                end='JAN-27-13')
+            'NYSE:F',
+            start='JAN-01-10',
+            end='JAN-27-13')
         assert np.issubdtype(data.Open.dtype, np.number)
         assert np.issubdtype(data.Close.dtype, np.number)
         assert np.issubdtype(data.Low.dtype, np.number)
@@ -167,9 +170,9 @@ def test_unicode_date(self):
         # see gh-8967
 
         data = web.get_data_google(
-                'NYSE:F',
-                start='JAN-01-10',
-                end='JAN-27-13')
+            'NYSE:F',
+            start='JAN-01-10',
+            end='JAN-27-13')
         assert data.index.name == 'Date'
 
     @skip_on_exception(RemoteDataError)
 
